test/fulfillment_order_test.rb

```diff
@@ -139,5 +139,223 @@ def setup
+        assert_equal 3, unsubmitted_fulfillment_order.id
+        assert_equal 'open', unsubmitted_fulfillment_order.status
+        assert_equal 'unsubmitted', unsubmitted_fulfillment_order.request_status
+      end
+
+      should "make a fulfillment request for a fulfillment order excluding unsubmitted" do
+        fake_original_fulfillment_order = ActiveSupport::JSON.decode(load_fixture('fulfillment_order'))
+        fake_original_fulfillment_order['status'] = 'closed'
+        fake_submitted_fulfillment_order = fake_original_fulfillment_order.clone
+        fake_submitted_fulfillment_order['id'] = 2
+        fake_submitted_fulfillment_order['status'] = 'open'
+        fake_submitted_fulfillment_order['request_status'] = 'submitted'
+        body = {
+          original_fulfillment_order: fake_original_fulfillment_order,
+          submitted_fulfillment_order: fake_submitted_fulfillment_order,
+          unsubmitted_fulfillment_order: nil,
+        }
+        request_body = {
+          fulfillment_request: {
+            fulfillment_order_line_items: [
+              { id: 1, quantity: 1 }
+            ],
+            message: 'Fulfill this FO, please.'
+          }
+        }
+        fake "fulfillment_orders/519788021/fulfillment_request", :method => :post,
+          :request_body => ActiveSupport::JSON.encode(request_body),
+          :body => ActiveSupport::JSON.encode(body)
+
+        fulfillment_order = ShopifyAPI::FulfillmentOrder.find(519788021)
+        params = {
+          fulfillment_order_line_items: [{ id: 1, quantity: 1 }],
+          message: "Fulfill this FO, please."
+        }
+        response_fulfillment_orders = fulfillment_order.request_fulfillment(params)
+
+        assert_equal 'closed', fulfillment_order.status
+        assert_equal 3, response_fulfillment_orders.size
+
+        original_fulfillment_order = response_fulfillment_orders['original_fulfillment_order']
+        assert_equal 519788021, original_fulfillment_order.id
+        assert_equal 'closed', original_fulfillment_order.status
+
+        submitted_fulfillment_order = response_fulfillment_orders['submitted_fulfillment_order']
+        assert_equal 2, submitted_fulfillment_order.id
+        assert_equal 'open', submitted_fulfillment_order.status
+        assert_equal 'submitted', submitted_fulfillment_order.request_status
+
+        assert_nil response_fulfillment_orders['unsubmitted_fulfillment_order']
+      end
+    end
+
+    context "#accept_fulfillment_request" do
+      should "accept a fulfillment request for a fulfillment order" do
+        fulfillment_order = ShopifyAPI::FulfillmentOrder.find(519788021)
+
+        message = "LGTM. Accept this FO fulfillment request"
+        request_body = {
+          'fulfillment_request' => {
+            'message' => message
+          }
+        }
+        fake_response = { fulfillment_order: fulfillment_order.attributes.merge(status: 'in_progress') }
```

Comment:
nit: the request_status would be changed to `accepted` as well as the status change.